torch/csrc/cudnn/Conv.cpp

```diff
@@ -225,6 +225,14 @@ void cudnn_convolution_forward(
   size_t workspaceSize;
   CHECK(cudnnGetConvolutionForwardWorkspaceSize(handle, idesc.desc, wdesc.desc,
       cdesc.desc, odesc.desc, fwdAlg, &workspaceSize));
+  size_t memFree;
+  size_t memTotal;
+  cudaMemGetInfo(&memFree, &memTotal);
```

Comment:
Yeah that's what I've been thinking. We should tell the cuDNN what's the upper bound on the workspace size and let it pick an algorithm that fits into memory.